CHAID/tree.py

```diff
@@ -226,6 +229,38 @@ def classification_rules(self, node=None, stack=None):
         else:
             return self.classification_rules(self.get_node(node.parent), stack)
 
+    def tree_predictions(self):
+        """
+        Calculates the row criteria that give rise
+        to a particular terminal node
+        """
+        tree_predictions = pd.DataFrame()
+        for node in self:
+            if node.is_terminal:
+                sliced_arr = np.array([x.original_vector for x in self.vectorised_array]).T[node.indices]
+                unique_set = np.vstack({ tuple(row) for row in sliced_arr })
+                index = pd.MultiIndex.from_arrays(np.transpose(unique_set))
+                if tree_predictions.empty:
+                    tree_predictions = pd.DataFrame([[node.node_id, node.predict]] * len(index), index=index)
+                else:
+                    tree_predictions = tree_predictions.append(pd.DataFrame([[node.node_id, node.predict]] * len(index), index=index))
+        tree_predictions.columns = ['node_id', 'prediction']
+        # need to retroactively fill missing values
```

Comment:
dataset may not contain all the other i variables at that node, and thus they will be missed off of the node